diff --git a/datafusion/catalog-listing/src/table.rs b/datafusion/catalog-listing/src/table.rs
index 3ff86662f..c9c952c08 100644
--- a/datafusion/catalog-listing/src/table.rs
+++ b/datafusion/catalog-listing/src/table.rs
@@ -693,13 +693,7 @@ impl ListingTable {
         {
             Some(Err(e)) => log::debug!("failed to split file groups by statistics: {e}"),
             Some(Ok(new_groups)) => {
-                if new_groups.len() <= file_group_count {
-                    partitioned_file_lists = new_groups;
-                } else {
-                    log::debug!(
-                        "attempted to split file groups by statistics, but there were more file groups than target_partitions; falling back to unordered"
-                    )
-                }
+                partitioned_file_lists = new_groups;
             }
             None => {} // no ordering required
         }
